### ISSUE TITLE
Injecting Configs with SENSE Orchestrator

### ISSUE THREAD

**Issue by PannuMuthu** [esnet/sense-rtmon#21]
Goal: Use SENSE orchestrator to inject user-level config parameters directly to real-time per-flow monitoring system (instead of having users enter it to the config file) 

Below is my start on a config file using only data from 3 types of sources:

1.  User-defined data (e.g. names of dashboards, flows, API keys, etc)
2. Recoverable data from subprocesses run through the SENSE orchestrator (e.g. NMAP, if config, etc)
3. Existing data pre-defined within the SENSE orchestrator

[config_orchestrator.txt](https://github.com/esnet/sense-rtmon/files/9455790/config_orchestrator.txt)

**Comment by xi-yang** [comment]
Firstly let's review the switchData. @PannuMuthu let me know if you agree with the assessment or not.

```
switchData:
  # User-defined
  job_name: 'snmp1'
  # The IP of the host which is running the SNMP Exporter container
  # User Defined
  SNMPHostIP: 198.32.43.16
  # Scrape interval/duration time
  scrapeInterval: 15s
  scrapeDuration: 5h
  # Retrievable using ifconfig
  target: 172.16.1.1
  # switch if format: " - IF: optional"
  switchif: ""
  # User-Defined
  #if-mib and F10-IF-EXTENSION-MIB
  params: ['if_mib']
  portIn: 
    ifIndex:
    # ifName: 'hundredGigE 1/27'
    vlan: 1000
    # vlanName: 'Vlan 1000'
  portOut: 
    ifIndex:
    # ifName: 'hundredGigE 1/31'
    vlan: 1000
    # vlanName: 'Vlan 1000'
```

*   job_name: 'snmp1'
   * This can local config that the Orchestrator does not provide. 

* SNMPHostIP: 198.32.43.16
   * Orchestrator needs to know about the host IP or name so it can dispatch instructions to where the agent is residing. 
   * But do we need it in the config (instructions) passed to the agent?

* Scrape interval/duration time scrapeInterval and scrapeDuration
  * Optional
  * There should be defaults for these if the Orchestrator does not specify. 

* target: 172.16.1.1
  * Optional
  * The orchestrator may not know the actual address based on its model data. As each agent handles one device by default, it should have a configuration for the default target upon deployment.

*   params: ['if_mib']
  *  Which MIBs to use will be vendor / device specific and can be part of the configuration upon deployment. Orchestrator should not care

* portIn and portOut
  * We should take parameters that the Orchestrators sees in its model. For Layer-2 flow, it will typically be {portName, vlanTag}.
  * ifIndex is device SNMP specific that is not included in the Orchestrator model. Therefore, the agent (exporter) needs to map from {portName, vlanTag} into ifIndex. 
  * For some other vendor / devices, the flow may not be identified through mapping of {portName, vlanTag} into ifIndex. MIBs other than if-mib2 or even private MIBs may be required. 

To summarize, the bare-bone instructions for the Orchestrator to communicate to the exporter agent for collecting on a flow would be.
* A global global ID. Something like `'flow_gid': '5dbccaf5-491f-405e-9de0-8e1a424303e6'`
* A list (normally two) of {portName, vlanTag} that tell where to associate the data for this flow. 
* Some optional parameters with default preconfigs that can be overridden by the Orchestrator 

All other information will be either preconfigured with defaults (like target) or be derived (like ifIndex) from these information.

**Comment by xi-yang** [comment]
@PannuMuthu Register with your ESnet or LBL credential on https://sense-o.es.net:8443/StackV-web/portal/ through CILogon. Then I can walk you through the portal to show how the topology and service model data would look like on Orchestrator. 

Some high level information can be learned from my talk (first on the demos list) here: https://www.nersc.gov/research-and-development/superfacility/

**Comment by xi-yang** [comment]
Close this and start a new Flow IDSpec issue.

**Comment by PannuMuthu** [comment]
Re-opening issue as per action item 1.1 in #25:

Define a minimum JSON structure for SENSE-O

**Comment by PannuMuthu** [comment]
New Config Layout with Flow IDSpec info: [updated_config_peer.yml](https://github.com/esnet/sense-rtmon/blob/main/config_flow/updated_config_peer.yml)
